pkg/operator/generic_client.go

```diff
@@ -141,61 +140,46 @@ func (p *genericClient) ApplyOperatorStatus(ctx context.Context, fieldManager st
 		return fmt.Errorf("applyConfiguration must have a value")
 	}
 
+	for i, val := range desiredConfiguration.Conditions {
+		// desired configuration may contain empty string fields.
+		// However, they are persisted by API server as nil. This causes a hotloop as empty string and nil
+		// are not equal. We explicitly convert empty strings to nil to prevent hotloop.
+		// This should be safe, because explicitly setting empty string has no meaning.
+		if len(ptr.Deref(val.Message, "")) == 0 {
+			desiredConfiguration.Conditions[i].Message = nil
```

Comment:
Nit: better to make a copy of `desiredConfiguration` before updating it to avoid side effects.

Reply:
It seems that `DeepCopy` isn't generated for `desiredConfiguration`.

Reply:
As an alternative, you might find all the places where `ApplyOperatorStatus` is invoked and make sure both Message and Reason are set to `nil` before.